sql/dd/impl/bootstrap/bootstrap_ctx.h

```diff
@@ -110,6 +110,7 @@ class DD_bootstrap_ctx {
   // default_dd_system_storage_engine
   legacy_db_type m_actual_dd_engine = DB_TYPE_UNKNOWN;
   legacy_db_type m_did_dd_engine_upgrade_from = DB_TYPE_UNKNOWN;
+  legacy_db_type m_upgraded_dd_engine = DB_TYPE_UNKNOWN;
```

Comment:
Might be misleading not to have system tables in the variable name, as DD tables
and system tables are different things. Unfortunately, this will have to churn in
neighbor variables too. `m_upgrade_dd_system_engine`?

Likewise for the methods below